sherpa-onnx/jni/jni_java.cc

```diff
@@ -0,0 +1,414 @@
+// sherpa-onnx/jni/jni_java.cc
+// It is based on jni.cc, but with some changes
+// Copyright   2022-2023  by zhaoming
+// this will be used for the java environment
+// com.k2fsa.sherpaonnx.rcglib.OnlineRecognizer
+// It supports parallel stream decoding and each
+// stream will assign a uuid id.
+//
+
+#include "jni.h"  // NOLINT
+
+#include <strstream>
+#include <utility>
+
+#include <fstream>
+#include <map>
+#include <random>
+#include <sstream>
+#include "sherpa-onnx/csrc/macros.h"
+#include "sherpa-onnx/csrc/online-recognizer.h"
+#include "sherpa-onnx/csrc/onnx-utils.h"
+
+#define SHERPA_ONNX_EXTERN_C extern "C"
+#include <mutex>  // NOLINT
+namespace sherpa_onnx {
+
+static std::random_device rd;
+static std::mt19937 gen(rd());
+static std::uniform_int_distribution<> dis(0, 15);
+static std::uniform_int_distribution<> dis2(8, 11);
+
+static std::string GenerateUuid() {
+  // generate uuid for streams
+  std::stringstream ss;
+  int i;
+  ss << std::hex;
+  for (i = 0; i < 8; i++) {
+    ss << dis(gen);
+  }
+  ss << "-";
+  for (i = 0; i < 4; i++) {
+    ss << dis(gen);
+  }
+  ss << "-";
+  for (i = 0; i < 3; i++) {
+    ss << dis(gen);
+  }
+  ss << "-";
+  ss << dis2(gen);
+  for (i = 0; i < 3; i++) {
+    ss << dis(gen);
+  }
+  ss << "-";
+  for (i = 0; i < 12; i++) {
+    ss << dis(gen);
+  }
+  return ss.str();
+}
+
+class JavaRecognizer {
+  // the recognizer use std::map to manage streams based on its uuid,
+  // it can be used for parallel stream decode
+ public:
```

Comment:
Could you reformat your C++ file with `clang-format`?

You can use
```
pip install clang-format
```
to install `clang-format`.